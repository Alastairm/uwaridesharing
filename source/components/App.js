import React, { Component } from 'react';
import { Image, Navigator, StyleSheet, View } from 'react-native';

import UserForm from './UserForm.js'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});


export default class App extends Component {
  render() {
    return (
    <Navigator
      initialRoute={{component: UserForm}}
      renderScene={(route, navigator) => {
          return React.createElement(route.component, { navigator });
      }}
    />
  );
  }
}
