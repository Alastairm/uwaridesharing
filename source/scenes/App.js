import React from 'react';
import { Navigator } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

import UserForm from './UserForm.js';

EStyleSheet.build({
});

export default function App() {
  return (
    <Navigator
      initialRoute={{ component: UserForm }}
      renderScene={(route, navigator) => (
        <route.component navigator={navigator} {...route.props} />
        )
      }
    />
  );
}
