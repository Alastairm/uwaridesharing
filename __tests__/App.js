import 'react-native';
import React from 'react';
import App from '../source/scenes/App.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const app = renderer.create(
    <App />
  ).toJSON();
  expect(app).toMatchSnapshot();
});
