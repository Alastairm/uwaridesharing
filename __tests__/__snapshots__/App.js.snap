// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="auto"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scaleX": 1,
              },
              Object {
                "scaleY": 1,
              },
              Object {
                "rotate": "0deg",
              },
              Object {
                "skewX": "0deg",
              },
              Object {
                "skewY": "0deg",
              },
            ],
          },
          null,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": 1334,
          }
        }
        virtual={undefined}
      >
        <Image
          resizeMode={undefined}
          source={
            Object {
              "uri": "https://fleeteng-static.s3.amazonaws.com/assets/background-081a9027c8c821541bf2d7816138fc87.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentInset={
              Object {
                "bottom": 0,
              }
            }
            enableAutoAutomaticScroll={true}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            keyboardDismissMode="interactive"
            onScroll={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={0}
            showsVerticalScrollIndicator={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              }
            }
            virtual={undefined}
          >
            <View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "navy",
                      "fontSize": 88,
                      "marginTop": 26.68,
                      "textAlign": "center",
                    }
                  }
                >
                  UniRide
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "navy",
                      "fontSize": 24,
                      "textAlign": "center",
                    }
                  }
                >
                  UWA Rideshare
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 60,
                  }
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#faba12",
                      "borderRadius": 4,
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "opacity": 1,
                      "overflow": "hidden",
                      "width": 180,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#007aff",
                          "fontSize": 17,
                          "fontWeight": "500",
                          "textAlign": "center",
                        },
                        null,
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        null,
                      ]
                    }
                  >
                    Sign In
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 60,
                  }
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#303030",
                      "borderRadius": 4,
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "opacity": 1,
                      "overflow": "hidden",
                      "width": 180,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#007aff",
                          "fontSize": 17,
                          "fontWeight": "500",
                          "textAlign": "center",
                        },
                        null,
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        null,
                      ]
                    }
                  >
                    API Test
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 60,
                  }
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits="button"
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#303030",
                      "borderRadius": 4,
                      "flex": 1,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "opacity": 1,
                      "overflow": "hidden",
                      "width": 180,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#007aff",
                          "fontSize": 17,
                          "fontWeight": "500",
                          "textAlign": "center",
                        },
                        null,
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        null,
                      ]
                    }
                  >
                    Clear App Data
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </Image>
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-start",
                "margin": 20,
              }
            }
          >
            <View
              danger={false}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0,0,0,0.8)",
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "minHeight": 50,
                  "paddingBottom": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                }
              }
              success={false}
              virtual={undefined}
              warning={false}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                virtual={undefined}
              />
            </View>
          </View>
        </Modal>
        <Modal
          animationType="fade"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.4)",
                "flex": 1,
                "justifyContent": "flex-end",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "elevation": 4,
                  "height": NaN,
                  "opacity": 1,
                  "padding": 15,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#757575",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                virtual={undefined}
              />
              <RCTScrollView
                dataArray={Array []}
                dataSource={
                  ListViewDataSource {
                    "items": 0,
                  }
                }
                enableEmptySections={true}
                renderRow={[Function]}
                renderScrollComponent={[Function]}
                style={
                  Object {
                    "marginHorizontal": -15,
                    "marginTop": 15,
                  }
                }
                virtual={undefined}
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </Modal>
      </View>
    </View>
  </View>
</View>
`;
